app/backend/src/clj/teet/file/file_tx.clj

```diff
@@ -76,20 +76,30 @@
   another file with the same metadata."
   [db {id :db/id :as file}]
   (let [task-id (get-in (du/entity db id) [:task/_files 0 :db/id])
+
+        ;; Take latest versions of all files in task
         task-files (map first
                         (d/q '[:find ?f
                                :where
                                [?t :task/files ?f]
                                [(missing? $ ?f :meta/deleted?)]
+                               (not-join [?f]
+                                         [?replacement :file/previous-version ?f])
                                :in $ ?t]
                              db task-id))
         old-file (d/pull db modify-file-keys id)
-        new-file (-> file
-                     (select-keys modify-file-keys)
-                     ;; Take :db/id from map as the ref attribute value
-                     (update :file/part :db/id)
-                     cu/without-nils)
-        metadata (file-db/file-metadata (du/entity db id new-file))]
+        new-file (select-keys file modify-file-keys)
+
+        ;; Fetch info needed to calculate metadata and
+        ;; make the changes to it
```

Comment:
The changes made could be explained here. Also, `pull-file-metadata-by-id` docstring says it "pull[s] all info that is needed", but here we pull some more. So some context for the extra data would be nice.

Reply:
It's the "make changes to it" part. The pull fetches what is already stored in database, but we want the metadata as it would be if we applied the change... that's why we pull the document group info that is sent by client.